### PR TITLE
Convert field string to enum.

### PR DESCRIPTION
**Pull request by crobibero**
Part of #4214 

~~Note: `GetProgramsDto.Fields` cannot be converted~~

### CODE REVIEW

File: Jellyfin.Api/Controllers/ArtistsController.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ public class ArtistsController : BaseJellyfinApiController
             [FromQuery] int? limit,
             [FromQuery] string? searchTerm,
             [FromQuery] string? parentId,
-            [FromQuery] string? fields,
+            [FromQuery] ItemFields[] fields,
```
Comment (nielsvanvelzen) [inline Jellyfin.Api/Controllers/ArtistsController.cs:89]: I'm slightly annoyed by the fact the type is called `ItemFields` instead of `ItemField` but I guess that's not in the scope of this PR.

**Comment by cvium** [comment]
> Note: `GetProgramsDto.Fields` cannot be converted

Why?

**Comment by crobibero** [comment]
> > Note: `GetProgramsDto.Fields` cannot be converted
> 
> Why?

It probably  could be converted, but we would need a new JsonConverter specific for that item because it’s currently passed as a comma delimited string instead of an array

**Comment by crobibero** [comment]
@cvium the `Fields` parameter is now converted using the `JsonCommaDelimitedArrayConverterFactory`.
Thing to note- make sure to import `System.Text.Json.Serialization` instead of `Newtonsoft.Json` or the converter won't work 🙃

File: Jellyfin.Api/Extensions/DtoExtensions.cs [hunk 2]
```diff
@@ -21,31 +21,11 @@ public static class DtoExtensions
         /// Legacy order: 1.
         /// </remarks>
         /// <param name="dtoOptions">DtoOptions object.</param>
-        /// <param name="fields">Comma delimited string of fields.</param>
+        /// <param name="fields">Array of item fields.</param>
         /// <returns>Modified DtoOptions object.</returns>
-        internal static DtoOptions AddItemFields(this DtoOptions dtoOptions, string? fields)
+        internal static DtoOptions AddItemFields(this DtoOptions dtoOptions, ItemFields[] fields)
```
Comment (Bond-009) [inline Jellyfin.Api/Extensions/DtoExtensions.cs:26]: We can remove this function now, no?

File: Jellyfin.Api/Models/LiveTvDtos/GetProgramsDto.cs [hunk 3]
```diff
@@ -167,6 +168,8 @@ public class GetProgramsDto
         /// Gets or sets specify additional fields of information to return in the output. This allows multiple, comma delimited. Options: Budget, Chapters, DateCreated, Genres, HomePageUrl, IndexOptions, MediaStreams, Overview, ParentId, Path, People, ProviderIds, PrimaryImageAspectRatio, Revenue, SortName, Studios, Taglines.
         /// Optional.
         /// </summary>
-        public string? Fields { get; set; }
+        [JsonConverter(typeof(JsonCommaDelimitedArrayConverterFactory))]
+        [SuppressMessage("Microsoft.Performance", "CA1819:ReturnArrays", MessageId = "Fields", Justification = "Imported from ServiceStack")]
+        public ItemFields[] Fields { get; set; } = Array.Empty<ItemFields>();
```
Comment (Bond-009) [inline Jellyfin.Api/Models/LiveTvDtos/GetProgramsDto.cs:173]: Can we make this a `IReadOnlyList` now?

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Review by Bond-009 — APPROVED** [review]

**Review by BaronGreenback — APPROVED** [review]

File: Jellyfin.Api/Controllers/ChannelsController.cs [hunk 4]
```diff
@@ -184,8 +183,8 @@ public ActionResult<ChannelFeatures> GetChannelFeatures([FromRoute, Required] st
         /// <param name="userId">Optional. User Id.</param>
         /// <param name="startIndex">Optional. The record index to start at. All items with a lower index will be dropped from the results.</param>
         /// <param name="limit">Optional. The maximum number of records to return.</param>
-        /// <param name="filters">Optional. Specify additional filters to apply.</param>
-        /// <param name="fields">Optional. Specify additional fields of information to return in the output. This allows multiple, comma delimited. Options: Budget, Chapters, DateCreated, Genres, HomePageUrl, IndexOptions, MediaStreams, Overview, ParentId, Path, People, ProviderIds, PrimaryImageAspectRatio, Revenue, SortName, Studios, Taglines.</param>
+        /// <param name="filters">Optional. Specify additional filters to apply. This allows multiple, comma delimited. Options: IsFolder, IsNotFolder, IsUnplayed, IsPlayed, IsFavorite, IsResumable, Likes, Dislikes.</param>
```
Comment (cvium) [inline Jellyfin.Api/Controllers/ChannelsController.cs:186]: Why did you add the second sentence but removed it elsewhere?

Comment (crobibero) [reply Jellyfin.Api/Controllers/ChannelsController.cs]: I believe this was the result of a bad merge
